### PR TITLE
[Merged by Bors] - Apply `WindowDescriptor` settings in all modes

### PR DESCRIPTION
**Pull request by devil-ira**
# Objective
Some settings were only applied in windowed mode.
Fix the issue in #6933 

# Solution
Always apply the settings.

### CODE REVIEW

**Review by hymm — APPROVED** [review]

**Review by NiklasEi — COMMENTED** [review]
This makes sense, just one question, since I currently cannot test this myself:
What does borderless fullscreen look like with decorations/resizeable? Is it no longer borderless?

File: crates/bevy_winit/src/winit_windows.rs [hunk 1]
```diff
@@ -63,18 +63,20 @@ impl WinitWindows {
...
+            WindowMode::Windowed => {
                 if let Some(sf) = scale_factor_override {
                     winit_window_builder.with_inner_size(logical_size.to_physical::<f64>(sf))
                 } else {
                     winit_window_builder.with_inner_size(logical_size)
                 }
             }
+        };
+
+        winit_window_builder = winit_window_builder
```
Comment (IceSentry) [inline crates/bevy_winit/src/winit_windows.rs:75]: I guess it's not an issue, but it seems to me like most of those settings simply won't do anything when not in windowed mode

Comment (devil-ira) [reply crates/bevy_winit/src/winit_windows.rs:75]: `always_on_top` and `transparent` work in all modes.
The `resizable` and `decorations` settings will still take effect if the window is switched to windowed mode later on.

**Comment by devil-ira** [comment]
> What does borderless fullscreen look like with decorations/resizable? Is it no longer borderless?

Good question :)
It's still borderless. The decorations/resizable settings only take effect in windowed mode :)

File: crates/bevy_winit/src/winit_windows.rs (hunk 1 above)
Comment (IceSentry) [reply crates/bevy_winit/src/winit_windows.rs:75]: Ah, right, didn't think about switching mode, makes sense.

**Review by IceSentry — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**